src/app/qgisapp.cpp

```diff
@@ -2230,12 +2244,56 @@ void QgisApp::createMenus()
   mWebMenu = new QMenu( tr( "&Web" ), menuBar() );
   mWebMenu->setObjectName( QStringLiteral( "mWebMenu" ) );
 
+
   // Help menu
   // add What's this button to it
   QAction *before = mActionHelpAPI;
   QAction *actionWhatsThis = QWhatsThis::createAction( this );
   actionWhatsThis->setIcon( QgsApplication::getThemeIcon( QStringLiteral( "/mActionWhatsThis.svg" ) ) );
   mHelpMenu->insertAction( before, actionWhatsThis );
+
+  this->createProfileMenu();
+}
+
+void QgisApp::refreshProfileMenu()
+{
+  mConfigMenu->clear();
+  QgsUserProfile *profile = userProfileManager()->userProfile();
+  mConfigMenu->setTitle( tr( "&User Profile: %1" ).arg( profile->alias() ) );
+
+  Q_FOREACH ( const QString &name, userProfileManager()->allProfiles() )
+  {
+    profile = userProfileManager()->profileForName( name );
```

Comment:
This is a bit messy - profileForName returns a new object, which is getting leaked here, vs the earlier assigned value of userProfile() which doesn't transfer ownership